api/base/caller.go

```diff
@@ -130,16 +131,58 @@ type facadeCaller struct {
 	facadeName  string
 	bestVersion int
 	caller      APICaller
+	tracer      coretrace.Tracer
 }
 
 var _ FacadeCaller = facadeCaller{}
 
+// NewFacadeCaller wraps an APICaller for a given facade name and the
+// best available version.
+func NewFacadeCaller(caller APICaller, facadeName string, options ...Option) FacadeCaller {
+	return NewFacadeCallerForVersion(caller, facadeName, caller.BestFacadeVersion(facadeName), options...)
+}
+
+// NewFacadeCallerForVersion wraps an APICaller for a given facade
+// name and version.
+func NewFacadeCallerForVersion(caller APICaller, facadeName string, version int, options ...Option) FacadeCaller {
+	fc := facadeCaller{
+		facadeName:  facadeName,
+		bestVersion: version,
+		caller:      caller,
+		tracer:      coretrace.NoopTracer{},
+	}
+
+	for _, option := range options {
+		fc = option(fc)
+	}
+
+	return fc
+}
+
 // FacadeCall will place a request against the API using the requested
 // Facade and the best version that the API server supports that is
 // also known to the client. (id is always passed as the empty string.)
-func (fc facadeCaller) FacadeCall(request string, params, response interface{}) error {
+func (fc facadeCaller) FacadeCall(request string, params, response interface{}) (err error) {
+	// context.TODO() is used here because the context isn't currently passed
+	// in via the FacadeCall. Once that's done, we can use the context that's
+	// passed in.
+	ctx := coretrace.WithTracer(context.TODO(), fc.tracer)
+
+	// The following trace is used to track the call to the facade.
+	ctx, span := coretrace.Start(ctx, coretrace.NameFromFunc(), coretrace.WithAttributes(
```

Comment:
In the (near) future when we pass the ctx through the `FacadeCall`s and therefore via the api client, where will we finally start this trace? is it always on the worker or the command that's making the api call? maybe there aren't any rules and we have to decide this on a case by case basis?